src/Nox.Lib/Middlewares/RelatedEntityRoutingMiddleware.cs

```diff
@@ -177,6 +185,23 @@ private bool IsFirstPairValid(string entityName, string navigationName)
         return _canRedirect.Contains((entityName, navigationName));
     }
 
+    private bool IsChainValid(IReadOnlyList<string> segments, int segmentCount)
+    {
+        var count = segmentCount;
+        if (segments[count - 1].Equals(_refSegment))
+            count--;
+
+        var navigationPropertiesCount = (count - 3) / 2;
```

Comment:
please add a comment to explain this.. in a few weeks we will not understand what we were doing :)